store/store.go

```diff
@@ -561,20 +561,25 @@ func LoadBlockStoreState(db dbm.DB) cmtstore.BlockStoreState {
 }
 
 // LoadTxInfo loads the TxInfo from disk given its hash.
-func (bs *BlockStore) LoadTxInfo(txHash []byte) *cmtstore.TxInfo {
+// Returns nil if the transaction is not found.
+func (bs *BlockStore) LoadTxInfo(txHash []byte) (*cmtstore.TxInfo, error) {
+	if len(txHash) == 0 {
+		return nil, fmt.Errorf("cannot load tx info with empty hash")
+	}
+
 	bz, err := bs.db.Get(calcTxHashKey(txHash))
 	if err != nil {
-		panic(err)
+		return nil, fmt.Errorf("failed to get tx info from db: %w", err)
```

Comment:
[question] should we add a unit test to check that this error is returned in certain cituations?